ibm/service/power/data_source_ibm_pi_volume_group.go

```diff
@@ -6,74 +6,77 @@ package power
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
-
 	"github.com/IBM-Cloud/power-go-client/clients/instance"
-	"github.com/IBM-Cloud/power-go-client/helpers"
 	"github.com/IBM-Cloud/power-go-client/power/models"
 	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
 func DataSourceIBMPIVolumeGroup() *schema.Resource {
 	return &schema.Resource{
 		ReadContext: dataSourceIBMPIVolumeGroupRead,
 		Schema: map[string]*schema.Schema{
-			PIVolumeGroupID: {
-				Type:         schema.TypeString,
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
 				Required:     true,
-				Description:  "ID or Name of the volume group",
+				Type:         schema.TypeString,
 				ValidateFunc: validation.NoZeroValues,
 			},
-			helpers.PICloudInstanceId: {
-				Type:         schema.TypeString,
+			Arg_VolumeGroupID: {
+				Description:  "The ID of the volume group.",
 				Required:     true,
+				Type:         schema.TypeString,
 				ValidateFunc: validation.NoZeroValues,
 			},
 
-			// Computed Attributes
-			"volume_group_name": {
-				Type:        schema.TypeString,
+			// Attributes
+			Attr_ConsistencyGroupName: {
 				Computed:    true,
-				Description: "Volume group name",
-			},
-			"status": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Description: "The name of consistency group at storage controller level.",
+				Type:        schema.TypeString,
 			},
-			"replication_status": {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_ReplicationStatus: {
+				Computed:    true,
+				Description: "The replication status of volume group.",
+				Type:        schema.TypeString,
 			},
-			"consistency_group_name": {
-				Type:     schema.TypeString,
-				Computed: true,
+			Attr_Status: {
+				Computed:    true,
+				Description: "The status of the volume group.",
+				Type:        schema.TypeString,
 			},
-			"status_description_errors": vgStatusDescriptionErrors(),
-		},
-	}
-}
-
-func vgStatusDescriptionErrors() *schema.Schema {
-	return &schema.Schema{
-		Type:     schema.TypeSet,
-		Computed: true,
-		Elem: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"key": {
-					Type:     schema.TypeString,
-					Computed: true,
-				},
-				"message": {
-					Type:     schema.TypeString,
-					Computed: true,
-				},
-				"volume_ids": {
-					Type:     schema.TypeList,
-					Computed: true,
-					Elem:     &schema.Schema{Type: schema.TypeString},
+			Attr_StatusDescriptionErrors: {
+				Computed:    true,
+				Description: "The status details of the volume group.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
Why not use the `vgStatusDescriptionErrors()` If it is used in other resource/datasource?

Reply:
The intention was to take those out, but something might have gotten muddled in the refactor.

Reply:
I think the function couldn't be fully removed without making changes to the resource also.

Reply:
I am still unclear why we are not re-using `vgStatusDescriptionErrors`? Seems the schema definition is same.